NBitcoin/BIP158/GolombRiceFilter.cs

```diff
@@ -180,20 +194,33 @@ public bool MatchAny(byte[][] data, byte[] key)
 		/// <param name="key">Key used for hashing the data elements.</param>
 		/// <returns>true if at least one of the elements is in the filter, otherwise false.</returns>
 		public bool MatchAny(IEnumerable<byte[]> data, byte[] key)
+		{
+			var reader = new GRCodedStreamReader(new BitStream(Data), P, 0);
+			return MatchAny(data, key, reader);
+		}
+
+		/// <summary>
+		/// Checks if any of the provided elements is in the filter.
+		/// </summary>
+		/// <param name="data">Data elements to check in the filter.</param>
+		/// <param name="key">Key used for hashing the data elements.</param>
+		/// <param name="reader">Golomb-Rice stream reader.</param>
+		/// <returns>true if at least one of the elements is in the filter, otherwise false.</returns>
+		public bool MatchAny(IEnumerable<byte[]> data, byte[] key, GRCodedStreamReader reader)
```

Comment:
I guess this can be static method?

Reply:
I think this method can be very useful in many scenarios. Just one example:

```c#
public void ProcessFilter(GolombRiceFilter filter, uint256 blockId, IEnumerable<Wallet> openedWallets)
{
   var reader = filter.GetGRStreamReader();
   foreach (var wallet in openedWallets)
   {
      if (filter.MatchAny(wallet.GetAllScriptPubKeys(), filterKey, reader))
      {
           Console.WriteLine($"The block {blockId} contains transactions relevant for wallet {wallet.Name}");
      }
   }
}
```